src/main/kotlin/calculator/Number.kt

```diff
@@ -0,0 +1,15 @@
+package calculator
+
+import java.lang.IllegalArgumentException
+
+data class Number constructor(
+    val value: Int
+) {
+    companion object {
+        fun from(value: String): Number = try {
+            Number(value.toInt())
+        } catch (_: NumberFormatException) {
+            throw IllegalArgumentException("\"$value\" 는 정수가 아닙니다.")
+        }
```

Comment:
runCatching을 활용해볼 수도 있겠네요 :)
```suggestion
        fun from(value: String): Number = runCatching { Number(value.toInt()) }
            .getOrNull()
            ?: throw IllegalArgumentException("\"$value\" 는 정수가 아닙니다.")
```

Reply:
runCatching을 사용하면 Result로 객체를 감싸서 에러 핸들링 처리 시점을 바꾸는거 같은데,
굳이 runCatching 활용 후 getOrNull()로 다시 꺼내어 사용하는 이유가 궁금합니다.

Reply:
핸들링 처리 시점을 바꾼다기 보다는, 더 코틀린스러운 코드를 작성하는 데에 더 초점이 맞춰져 있습니다.
내부에서 try catch를 사용하기 때문에 해당 블록을 줄일 수 있고, 함수형 패러다임의 로직을 작성할 수도 있습니다.

Reply:
`getOrNull()`를 사용한 시점에서 말씀해주신 장점들이 희석 되는게 아닌가 해서요...
함수형 패러다임의 로직을 작성한다는 건 예시를 부탁드려도 괜찮을까요?!!

Reply:
함수형 패러다임은 위 suggestion 로직처럼 체이닝한 것을 예시로 들었습니다.

코틀린에는 이런 유틸성 함수와 확장함수가 매우 많습니다.
.also 같은 스코프 함수들또한 내부를 들여다보면, 파라미터로 전달 받은 람다를 호출하고 자기 자신을 넘길 뿐입니다.
하지만 이런 기능들을 잘 조합하면 함수를 체이닝하면서 함수형 패러다임을 가져갈 수도 있고, 매우 코틀린스러운 로직을 작성할 수 있게 만들어줍니다.

Java 스타일 대로 try catch 블럭을 사용하셔도 무방합니다. :)